### PR TITLE
KAN-206 - Create shared/reusable workflow 

### PR DESCRIPTION
**Pull request by andriiholubenko**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          pnpmVersion: ${{inputs.pnpmVersion}}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.natToken }}
+          NODE_OPTIONS: ${{ inputs.nodeOptions }}
+
+  build:
+    runs-on: self-hosted
+    needs: dependencies
+    outputs:
+      builded: ${{ steps.build.outputs.builded }}
```
Comment (vtereshyn) [inline .github/workflows/build.yml:54]: why do you need this?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+on:
+  workflow_call:
+    inputs:
+      nodeVersion:
+        required: true
```
Comment (vtereshyn) [inline .github/workflows/build.yml:5]: probably should be `false`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      pnpmVersion:
+        type: string
+        required: false
+        default: 7.13.3
+      nodeOptions:
+        type: string
+        required: false
+        default: --max_old_space_size=8192 
+    secrets:
+      natToken:
```
Comment (vtereshyn) [inline .github/workflows/build.yml:28]:
```suggestion
      NODE_AUTH_TOKEN:
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      name:
+        required: false
+        type: string
+      packageManager:
+        type: string
+        required: false
+        default: yarn
+      cacheVersion:
+        type: string
+        required: false
```
Comment (vtereshyn) [inline .github/workflows/build.yml:17]: should be true and let's put v1 here

Comment (vtereshyn) [reply .github/workflows/build.yml:17]: each repo that's gonna use this `shared-workflow` should pass worn cache version

**Review by vtereshyn — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+          sudo apt update
+          sudo apt install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xvfb
+
+      - uses: AstrumU/shared-workflows/setup@feature/KAN-206
+        name: setup env
+        with:
+          nodeVersion: ${{inputs.nodeVersion}}
+          packageManager: ${{inputs.packageManager}}
+          cacheVersion: ${{inputs.cacheVersion}}
+          pnpmVersion: ${{inputs.pnpmVersion}}
```
Comment (vtereshyn) [inline .github/workflows/tests.yml:123]:
```suggestion
          nodeVersion: ${{ inputs.nodeVersion }}
          packageManager: ${{ inputs.packageManager }}
          cacheVersion: ${{ inputs.cacheVersion }}
          pnpmVersion: ${{ inputs.pnpmVersion }}
```

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+          key: ${{ github.sha }} 
+
+      - name: Install & verify Cypress
+        run: |
+          yarn cypress install
+          yarn cypress verify
+          yarn cypress info
+          yarn cypress version
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NODE_AUTH_TOKEN }}
```
Comment (vtereshyn) [inline .github/workflows/tests.yml:142]:
```suggestion
          NODE_AUTH_TOKEN: ${{ secrets.NODE_AUTH_TOKEN }}
```

why are we using `NODE_AUTH_TOKEN` here but not `natToken`?

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+          pnpmVersion: ${{inputs.pnpmVersion}}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.natToken }}
+          NODE_OPTIONS: ${{ inputs.nodeOptions }}
+
+      - uses: actions/cache@v3
+        id: build-cache
+        name: Restore build cache
+        with:
+          path: ./*
```
Comment (vtereshyn) [inline .github/workflows/ci.yml:102]: should cache only `dist` folders

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        default: yarn
+      cacheVersion:
+        type: string
+        required: true
+        default: v1
+      pnpmVersion:
+        type: string
+        required: false
+        default: 7.13.3
```
Comment (vtereshyn) [inline .github/workflows/build.yml:22]: we agreed to put `7` here

**Review by yevhenii-r — CHANGES_REQUESTED** [review]
lets configure separate job in separate file

**Review by yevhenii-r — APPROVED** [review]

**Pull request merged**